Realm/RLMObjectSchema.mm

```diff
@@ -104,7 +102,7 @@ + (instancetype)schemaForObjectClass:(Class)objectClass {
     }
     schema.className = className;
     schema.objectClass = objectClass;
-    schema.accessorClass = RLMDynamicObject.class;
+    schema.accessorClass = objectClass;
```

Comment:
I suspect it was to ensure that an accessor class wouldn't get created for the object schema, back from when we only supported a single accessor class per name.